tests/test_intercept.js

```diff
@@ -1711,3 +1712,43 @@ test('teardown', t => {
   t.deepEqual(leaks, [], 'No leaks')
   t.end()
 })
+
+test('works when headers removed by http-proxy', t => {
```

Comment:
We're in the process of changing the way we test nock–client integration issues. These issues are important, though it's unwieldy to include large-bracket tests in this repo because it doesn't tell nock developers what the problem is. In addition, the internals of client libraries can change. While this is a decent regression test for the integration, it doesn't prevent against regression of the specific behaviors we want to ensure.

The idea is to build nock–client integration tests in separate repositories, which test the latest development version of nock against. We'd really welcome help getting that started! See a discussion in #1427 which goes into more detail about it.

What we need here is a unit test, probably using the `http` module, which exercises the issue you're seeing, in a way that a future developer working on nock can understand. Is it possible to express the bug that way?

In general we want to do the standard thing, not implement workarounds for libraries, so writing the test using `http` should help make it clear whether what we're doing here is sufficiently standard. (It sounds like it _is_, though TBH it's we're all still getting up to speed with the Nock codebase.)

Reply:
Fair enough - I think I can probably pull out what http-proxy is doing into the test itself and remove the dependency :).